crypto/rand/rand_lib.c

```diff
@@ -22,6 +22,14 @@
 #endif
 #include "e_os.h"
 
+#if defined(OPENSSL_SYS_UNIX) && !defined(HAVE_CLOCK_GETTIME)
+# if defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 199309L
+#   define HAVE_CLOCK_GETTIME 1
```

Comment:
From reading the manpage, you also should check _POSIX_TIMERS (and _POSIX_MONOTONIC_CLOCK)